src/argument.cpp

```diff
@@ -206,5 +208,18 @@ argument argument::element(std::size_t i) const
     return argument{shape{this->get_shape().type()}, this->data() + offset};
 }
 
+void save_argument(const argument& a, const std::string& filename)
+{
+    migraphx::value v = to_value(a);
+    write_buffer(filename, to_msgpack(v));
+}
+
+argument load_argument(const std::string& filename)
+{
+    migraphx::argument a;
+    from_value(from_msgpack(read_buffer(filename)), a);
+    return a;
```

Comment:
Write this as `return from_value<argument>(from_msgpack(read_buffer(filename)))`.